tools/setup_helpers/ninja_builder.py

```diff
@@ -86,7 +244,13 @@ def link(self, target_desc, objects,
                       export_symbols, debug, extra_preargs,
                       extra_postargs, build_temp, target_lang)
 
-        with patch(distutils.unixccompiler.UnixCCompiler, '_compile', _compile):
-            with patch(distutils.unixccompiler.UnixCCompiler, 'link', link):
-                with patch(self, 'force', True):
-                    self._build_default(ext)
+        if os.sys.platform == 'win32':
+            with patch(distutils.msvccompiler.MSVCCompiler, 'compile', win_compile):
```

Comment:
Consider reducing copy paste here by doing something like:

```
if os.sys.platform == 'win32':
  compiler = distutils.msvccompiler.MSVCCompiler
  new_compile, new_link = win_compile, win_link
else:
  compiler = distutils.unixccompiler.UnixCCompiler
  new_compile, new_link = unix_compile, new_link
# ...etc..
```

Reply:
They're not the same. `win_compile` overrides `compile` but `unix_compile` overrides `_compile`.